pkg/core/native/native_neo.go

```diff
@@ -437,6 +463,21 @@ func (n *NEO) PostPersist(ic *interop.Context) error {
 			}
 		}
 	}
+	// Update next block validators cache for external users.
+	var (
+		h        = ic.Block.Index // consider persisting block as stored to get _next_ block validators
+		numOfCNs = n.cfg.GetNumOfCNs(h + 1)
+	)
+	if cache.validators == nil || numOfCNs != len(cache.validators) {
```

Comment:
`validators` can't be `nil`, so `votesChanged` can be checked.

Reply:
We have a single case with `unregister` where `votesChanged` isn't set to `true`. Fixed.